MWT/architecture.rst

```diff
@@ -0,0 +1,99 @@
+   :align: center
+
+Widget
+~~~~~~
+
+A widget is an object that is intended to be displayed on a screen and that can interact with the user. A Widget occupies a specific region of the display and holds a state.
+
+Widgets are arranged on a desktop. A widget can be part of only one desktop hierarchy, and can appear only once on that desktop.
+
+Container
+~~~~~~~~~
+
+A container follows the composite pattern: it is a widget composed of other widgets. It also defines the layout policy of its children (defining their bounds). The children's locations are relative to the location of their parent.
+Containers can be nested to design elaborate user interfaces.
+
+Widgets added to a container are stored in a list. The order of the list defines the widgets' front-to-back stacking order within the container. By default, a Widget is added to the end of the list (so to the bottom of the stacking order).
+
+Desktop
+~~~~~~~
+
+A desktop is a displayable intended to be shown on a display (cf. MicroUI). At any time, only one desktop can be displayed per display.
+
+A desktop contains a widget (or a container). When the desktop is shown, its widget (and all its hierarchy for a container) is drawn on the display.
+
+Rendering
+---------
+
+A new rendering of a widget on the display can be requested by calling its `requestRender()` method. The rendering is done asynchronously in the MicroUI thread.
+
+When a container is rendered, all its children are also rendered.
+
+A widget can be transparent, meaning that it does not fill its whole bounds. In this case, when this widget is asked to be rendered, its parent is asked to be rendered in the area of the widget (recursively if the parent is also transparent). Usually a widget is transparent when its background (from the style) is transparent.
+
+A widget can also be rendered directly in a specific graphics context by calling its `render(GraphicsContext)` method. It can be useful to render a widget (and its children) in an image for example.
+
+Lay Out
+-------
+
+All widgets are laid out at once during the lay out process. This process can be started by `Desktop.requestLayOut()` and `Widget.requestLayOut()`. This process is composed of two steps, each step browses the hierarchy of widgets following a depth-first algorithm:
+* compute the optimal size for each widget and container (considering the constraints of the lay out),
+* set location and size for each widget.
+
+Once the location and size of a widget is set, the widget is notified by a call to `onLaidOut()`.
+
+Event Dispatch
+--------------
+
+Events generated in the hardware (touch, buttons, etc.) are sent to the event dispatcher of the desktop. It is then responsible of sending the event to one or several widgets of the hierarchy. A widget receives the event through its `handleEvent(int)` method. This method returns a boolean that indicates whether or not the event has been consumed by the widget or not.
+
+Disabled widgets cannot receive events.
```

Comment:
and widgets are disabled by default

Reply:
Fixed.

Reply:
your new sentence is weird:
`Disabled widgets cannot receive events and widget are disabled by default.`

Reply:
i think you can remove the extra `widget`

Reply:
Ok, I rephrase.